File: Battleships.Models/GameStateModel.cs
using System.ComponentModel.DataAnnotations;

namespace Battleships.Models
{
    public class GameStateModel
    {
        public int GameId { get; set; }
        public int GameTurnNumber { get; set; }
        public int GameTurnPlayer { get; set; }
        public int FireRow { get; set; }
        public int FireCol { get; set; }
        public bool GameMulti { get; set; }
        public bool GameOpen { get; set; }
        public int GameSpeedDivider { get; set; }
        [Required(ErrorMessage = "Please fill up.")]
        public string GameDifficulty { get; set; }
        public bool DisplayingResults { get; set; }
        public bool FireResult { get; set; }
        public Player[] Players { get; set; }
        public bool IsDeploymentAllowed { get; set; }
        public bool IsStartAllowed { get; set; }

    }
}

File: Battleships.AppClient/src/app/app-core/_models/game-state.model.ts
import { Player } from '@models/player.model';

export interface GameState {
  gameId: number;
  gameTurnNumber: number;
  gameTurnPlayer: number;
  fireRow: number;
  fireCol: number;
  gameMulti: boolean;
  gameOpen: boolean;
  gameSpeedDivider: number;
  gameDifficulty: string;
  displayingResults: boolean;
  fireResult: boolean;
  players: Player[];
  isDeploymentAllowed: boolean;
  isStartAllowed: boolean;
}
